test/btrust.test.js

```diff
@@ -0,0 +1,260 @@
+            const { v, r, s } = EIP712.sign(Domain(btrust), 'Delegation', { delegatee, nonce, expiry }, Types, signatory.privateKey);
+            expect(btrust.delegateBySig(delegatee, nonce, expiry, v, r, s)).to.eventually.be.rejectedWith("revert BTRUST::delegateBySig: signature expired");
+        });
+
+        it('delegates on behalf of the signatory', async () => {
+            const delegatee = owner, nonce = 0, expiry = 10e9;
+            const signatory = web3.eth.accounts.create();
+            const { v, r, s } = EIP712.sign(Domain(btrust), 'Delegation', { delegatee, nonce, expiry }, Types, signatory.privateKey);
+            const tx = await btrust.delegateBySig(delegatee, nonce, expiry, v, r, s);
+            expect(tx.gasUsed < 80000);
+            result = await btrust.delegates(signatory.address)
+            expect(result).to.equal(owner);
+        });
+    });
+
+    describe('numCheckpoints', () => {
+        it('returns the number of checkpoints for a delegate', async () => {
+            let guy = accounts[0];
+            await btrust.transfer(guy, '100', { from: owner })
+            result = await btrust.numCheckpoints(a1);
+            console.log(web3.utils.isBN(result))
+            expect(result.toString()).to.equal('0');
+
+            const t1 = await btrust.delegate(a1, { from: guy });
+            result = await btrust.numCheckpoints(a1);
+            expect(result.toString()).to.equal('1');
+
+            const t2 = await btrust.transfer(a2, 10, { from: guy });
+            result = await btrust.numCheckpoints(a1);
+            expect(result.toString()).to.equal('2');
+
+            const t3 = await btrust.transfer(a2, 10, { from: guy });
+            result = await btrust.numCheckpoints(a1);
+            expect(result.toString()).to.equal('3');
+
+            const t4 = await btrust.transfer(guy, 20, { from: owner });
+            result = await btrust.numCheckpoints(a1);
+            expect(result.toString()).to.equal('4');
+
+            result = await btrust.checkpoints(a1, 0);
+            expect(result.fromBlock.toString()).to.equal(t1.receipt.blockNumber.toString());
+            expect(result.votes.toString()).to.equal('100');
+
+            result = await btrust.checkpoints(a1, 1);
+            expect(result.fromBlock.toString()).to.equal(t2.receipt.blockNumber.toString());
+            expect(result.votes.toString()).to.equal('90');
+
+            result = await btrust.checkpoints(a1, 2);
+            expect(result.fromBlock.toString()).to.equal(t3.receipt.blockNumber.toString());
+            expect(result.votes.toString()).to.equal('80');
+
+            result = await btrust.checkpoints(a1, 3);
+            expect(result.fromBlock.toString()).to.equal(t4.receipt.blockNumber.toString());
+            expect(result.votes.toString()).to.equal('100');
+        });
+
+        it.skip('does not add more than one checkpoint in a block', async () => {

```

Comment:
plz remove .skip - don't want to merge it in with skipped test. how are you coming along with this test? still stuck?